### ISSUE TITLE
Function.function_type has no way to determine whether it was set by user

### ISSUE THREAD

**Issue by verylazyguy** [Vector35/binaryninja-api#3883]
**What is the feature you'd like to have?**
It would be nice to be able to know if a function's function_type was set by auto analysis (`set_auto_type`) or by the user (`set_user_type`)

**Is your feature request related to a problem?**
Yes. Currently this information is not available

**Are any alternative solutions acceptable?**
Unfortunately not

**Additional Information:**
the function's `auto` property could be repurposed for this. Is the information it gives useful for anything else?

**Comment by plafosse** [comment]
Yeah internal to the function object there is a `m_userTypeSet` boolean which isn't exposed.

**Comment by plafosse** [comment]
I've wanted this API as well, my workaround was to check the confidence of the return type was == 255 which is a pretty safe heuristic but not completely safe

**Comment by plafosse** [comment]
This api exists as of `3.4.4271` `Function.has_user_type`
